hazelcast/src/main/java/com/hazelcast/nio/ConnectionObservable.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.nio;
+
+import com.hazelcast.spi.annotation.PrivateApi;
+
+/**
+ * Provides connection listen capabilities.
+ */
+@PrivateApi
+public interface ConnectionObservable {
+
```

Comment:
Hmm, `observable` would imply that there's a `addObserver` method.
I would propose renaming it to `ConnectionListenable`